src/battle-scene.ts

```diff
@@ -3325,6 +3326,48 @@ export default class BattleScene extends SceneBase {
     }
   }
 
+  /**
+   * Checks if a {@linkcode Pokemon} can be released from party
+   * @param slotIndex the position of the Pokemon the player is trying to release
+   * @returns `true` if any of the remaining Pokemon in the party is {@linkcode Pokemon.isAllowedInBattle | allowed in battle}, otherwise `false`
+   */
+  canReleasePokemon(slotIndex: number): boolean {
+    const currentParty = this.getPlayerParty();
+    const postReleaseParty = this.getPlayerParty().filter((p) => currentParty[slotIndex].id !== p.id);
```

Comment:
NIT: Why not use the previously buffered `currentParty`?
```suggestion
    const postReleaseParty = currentParty.filter((p) => currentParty[slotIndex].id !== p.id);
```